### ISSUE TITLE
Look at existing Loopring web wallet and take best parts

### ISSUE THREAD

**Issue by lehighjcut** [dolomite-exchange/dolomite#21]
There already is a minimal version of what we're building made by Loopring themself and it's open sourced here https://github.com/Loopring/loopr. It's made in React, like ours will be. We need to do a couple things.

- [ ] 1) Decide if we should start from scratch and copy/paste in the best parts of their code OR should we start from their code and modify. This decision will have to wait until we get a full set of the required features from @coreycaplan3 and @aknuck 

- [ ] 2) Look at their project structure, learn from what seems to be nice about it and what doesn't seem to be nice about it. Copy or find the best parts and document what you find.

- [ ] 3) Look at the libraries they are using and see if we should use the same / add to our project.

- [ ] 4) Look at how they implement the crypto related interactions, like decrypting a wallet, making a trade, interacting with metamask, and try to copy those bits over into modularized chunks.

This issue will likely be linked to multiple PRs so just comment on the issue as they come. Make sure to use the wallet first to get an idea of what its like! https://loopr.io/#/wallet/assets

**Comment by zrubenst** [comment]
I think the most useful thing is the files in `src/modules/account` as these files handle the logging in from metamask/ledger/keystore/mnemonic/etc. and they all subclass an Account class and each offers the ability to sign transactions and commit transactions using the different methods of logging in. So we should take these classes and adapt them to follow our own class structures/coding style. We will create our own UI for this, because based on @aknuck's mocks I don't see a succinct way to adapt loopr UI components for accounts, but the services/modules for it will help us a lot

**Comment by lehighjcut** [comment]
Yes I agree I don't think we'll use their code for the UI much but
definitely for a lot of the boilerplate crypto transaction stuff that has
likely already been tested thoroughly on their end should help us a lot

**Comment by lehighjcut** [comment]
@zrubenst if you're good on this and don't need this open and feel you have sufficiently discussed with @coreycaplan3 / @aknuck then one of you guys close it
